src/main/java/forestry/core/FluidFogEventHandler.java

```diff
@@ -0,0 +1,51 @@
+package forestry.core;
+
+import forestry.core.config.Constants;
+import forestry.core.fluids.ForestryFluids;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+import net.minecraftforge.client.event.EntityViewRenderEvent;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+import java.awt.*;
+
+@OnlyIn(Dist.CLIENT)
+@Mod.EventBusSubscriber(modid = Constants.MOD_ID, value = Dist.CLIENT)
+public class FluidFogEventHandler {
+	@SubscribeEvent(priority = EventPriority.HIGH)
+	public static void onFogColorRender(EntityViewRenderEvent.FogColors event) {
+		if (event.getInfo().getFluidInCamera().getType() != Fluids.EMPTY)
```

Comment:
Wouldn't it make sense to create a "isForestryFluid" method and just tested against it. This would reduce the amount of replication by a lot